discovery/service.go

```diff
@@ -133,9 +123,13 @@ type AuthenticatedGossiper struct {
 	// reconstruct a full authenticated channel announcement. Once we
 	// receive the other half the channel proof, we'll be able to properly
 	// validate it an re-broadcast it out to the network.
-	//
-	// TODO(andrew.shvv) make this map persistent.
-	waitingProofs map[waitingProofKey]*lnwire.AnnounceSignatures
+	proofStore *channeldb.IndexStore
+
+	// proofsToIndex stores the correspodence of the waiting proof key
+	// and index of this object inside the storage itself. By using such
+	// approach we checking the existence of object without touching the db
+	// itself.
+	proofToIndex map[waitingProofKey]uint64
```

Comment:
If we do away with the `IndexStore` abstraction, then this map and all the code around managing it are no longer needed.